vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/SortingPage.java

```diff
@@ -52,6 +57,80 @@ public SortingPage() {
         });
         button.setId("sort-by-age");
         add(button);
+
+        setSizeFull();
+        gui1 = new TestGui();
+        Button btnReset = new Button("reset", evt -> gui1.getGrid().sort(null));
+        btnReset.setId("btn-reset");
+        Button btRm = new Button("deattach", evt -> remove(gui1));
+        btRm.setId("btn-deattach");
+        Button btattach = new Button("attach",
+                evt -> add(gui1));
+        btattach.setId("btn-attach");
+        add(btRm, btattach,btnReset);
+    }
+
+    public class TestGui extends FlexLayout {
```

Comment:
Why can't we use the existing grid and data for the new test cases?

Reply:
in the first time, I used the existing grid, but after fixed the bug (reattach and grid keeps the sort state, there is a new issue that the number of vaadin-grid-sorter is increased after detach/reattach-ing grid. and it is quite hard to assert if the grid has only 2 columns and we just have sort number state is 1-2 or 2-1. That is why I have to add a new grid to check the number like 1-2-3 or 1-3-2 or 3-1-2. But this morning, we agree that is another issue and we will separate it. The new issue is that if clicking re(attach) is more 2 times without detaching, the grid can not keep the sort state.

So I will back to work on the existing grid because I just check the grid keeps the sort state after the grid is reattached or refresh the page.

Reply:
If a third column is needed (not sure if it is), you can add a new column to the existing grid.

Reply:
I backed to work on the existing grid, edited tests with the existing grid.
And with this issue, we do not need the second grid.
in the future, when working on the issue that reattaches much time without detaching, maybe we can add new column for the test.